src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -203,6 +207,69 @@ func (r *Cluster) validateKafkaListeners() field.ErrorList {
 	return allErrs
 }
 
+func (r *Cluster) validatePandaproxyListeners() field.ErrorList {
+	var allErrs field.ErrorList
+	var proxyExternal *PandaproxyAPI
+	kafkaExternal := r.ExternalListener()
+	for i, p := range r.Spec.Configuration.PandaproxyAPI {
+		if !p.External.Enabled {
+			continue
+		}
+		if proxyExternal != nil {
+			allErrs = append(allErrs,
+				field.Invalid(field.NewPath("spec").Child("configuration").Child("pandaproxyApi"),
+					r.Spec.Configuration.PandaproxyAPI,
+					"only one pandaproxy api listener can be marked as external"))
+		}
+		proxyExternal = &r.Spec.Configuration.PandaproxyAPI[i]
+		if proxyExternal.Port != 0 {
+			allErrs = append(allErrs,
+				field.Invalid(field.NewPath("spec").Child("configuration").Child("pandaproxyApi"),
+					r.Spec.Configuration.PandaproxyAPI,
+					"external pandaproxy api listener cannot have port specified, it's autogenerated"))
+		}
+		if (kafkaExternal == nil || !kafkaExternal.External.Enabled) && (proxyExternal != nil && proxyExternal.External.Enabled) {
+			allErrs = append(allErrs,
+				field.Invalid(field.NewPath("spec").Child("configuration").Child("pandaproxyApi"),
+					r.Spec.Configuration.PandaproxyAPI,
+					"cannot have a pandaproxy external listener without a kafka external listener"))
+		}
+		if kafkaExternal == nil && proxyExternal.External.Subdomain != "" {
+			allErrs = append(allErrs,
+				field.Invalid(field.NewPath("spec").Child("configuration").Child("pandaproxyApi"),
+					r.Spec.Configuration.PandaproxyAPI,
+					"kafka external listener must be specified with the same sudomain as that of the external pandaproxy"))
+		}
+		if kafkaExternal != nil && kafkaExternal.External.Subdomain != proxyExternal.External.Subdomain {
+			allErrs = append(allErrs,
+				field.Invalid(field.NewPath("spec").Child("configuration").Child("pandaproxyApi"),
+					r.Spec.Configuration.PandaproxyAPI,
+					"sudomain of external pandaproxy must be the same as kafka's"))
+		}
+	}
+
+	// TLS for pandaproxy is not supported yet
+	for i, p := range r.Spec.Configuration.PandaproxyAPI {
+		if p.TLS.Enabled {
+			allErrs = append(allErrs,
+				field.Invalid(field.NewPath("spec").Child("configuration").Child("pandaproxyApi").Index(i).Child("tls"),
+					r.Spec.Configuration.PandaproxyAPI[i].TLS,
+					"no listener can have TLS enabled"))
+			// we need to run the validation on all listeners to also catch errors like !Enabled && RequireClientAuth
+			allErrs = append(allErrs, validatePandaproxyTLS(p.TLS, field.NewPath("spec").Child("configuration").Child("pandaproxyApi").Index(i).Child("tls"))...)
+		}
+	}
+
+	if !((len(r.Spec.Configuration.PandaproxyAPI) == 2 && proxyExternal != nil) || (proxyExternal == nil && len(r.Spec.Configuration.PandaproxyAPI) <= 1)) {
```

Comment:
uff it's hard to parse these conditions :D not your fault, just noting it here if you can come up with easier way to put this in. Maybe split it into two named functions? 🤔

Reply:
Agree, I'll leave this for a separate PR for all listeners.